tools/azsdk-cli/Azure.Sdk.Tools.Cli/Prompts/PromptTemplateBuilder.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.Sdk.Tools.Cli.Prompts;
+
+/// <summary>
+/// Builder for creating prompt contexts in a fluent manner.
+/// </summary>
+public class PromptContextBuilder
```

Comment:
I don't think I fully understand the value of having the builder here? If our prompts are taking in args why cant we just directly call our prompt template and pass in the args to be added into the overall string we are creating. Is this more .net design aligned?